templates/antora/template.js

```diff
@@ -0,0 +1,324 @@
+var YAML = require('js-yaml')
+var BASE_PATH = process.env.BASE_PATH || '/_data/antora';
+
+var AntoraTemplate = require('./antora.converter.js');
+// correlates to tinymce-docs antora path
+var AntoraNavBaseDir = 'apis/'; 
+
+var namespaceDescriptions = {
+  'tinymce': 'Global APIs for working with the editor.',
+  'tinymce.dom': 'APIs for working with the DOM from within the editor.',
+  'tinymce.editor.ui': 'APIs for registering User Interface components.',
+  'tinymce.geom': 'Various rectangle APIs.',
+  'tinymce.html': 'APIs for working with HTML within the editor.',
+  'tinymce.util': 'Browser related APIs.'
+};
+
+/**
+ * [function description]
+ * @param  {[type]} root   [description]
+ * @param  {[type]} toPath [description]
+ * @return {[type]}        [description]
+ */
+exports.template = function (root, toPath) {
+  var archive = new ZipWriter();
+  var rootTemplate = compileTemplate('root.handlebars');
+  var namespaceTemplate = compileTemplate('namespace.handlebars');
+  var template = compileTemplate('member.handlebars');
+
+  // bind new archive to function
+  addPageToArchive = addPageToArchive.bind(archive); // jshint ignore:line
+
+  // sort types alphabetically
+  var sortedTypes = root.getTypes().sort(function (a, b) {
+    if (a.fullName < b.fullName) return -1;
+    if (a.fullName > b.fullName) return 1;
+    return 0;
+  });
+
+  var YMLNav = getNavFile(sortedTypes);
+  var AdocNav = YMLNavToAdoc (YMLNav);
+
+  addPageToArchive({
+    filename: '_data/nav.yml',
+    content:  YAML.dump(YMLNav)
+
+  });
+
+  addPageToArchive({
+    filename: '_data/moxiedoc_nav.adoc',
+    content: AdocNav
+  });
+
+  // create all json and adoc for each item
+  var pages = sortedTypes.map(getMemberPages.bind(null, root, template))
+  var convertedPages = AntoraTemplate.convert(pages);
+  flatten(convertedPages).forEach(addPageToArchive);
+
+  getNamespacesFromTypes(sortedTypes).map(function (namespace) {
+    // TODO flatten FS here for antora if needed.
+    var fileName = (BASE_PATH + '/' + namespace + '.adoc').toLowerCase();
+    if (namespace in namespaceDescriptions) {
+      var namespaceDescription = namespaceDescriptions[namespace];
+    } else {
+      var namespaceDescription = namespace;
+    }
+    return {
+      filename: fileName,
+      content: namespaceTemplate({
+        title: namespace,
+        desc: namespaceDescription
+      })
+    };
+  }).forEach(addPageToArchive);
+
+  archive.saveAs(toPath, function (err) {
+    if (err) throw err;
+  });
+};
+
+function YMLNavToAdoc (navyml) {
+  // Api index page
+  var adoc = '* API Reference\n';
+  var pages = navyml[0].pages
+
+  // generate API namespaces
+  pages.forEach(function (namespace) {
+    // main namespace level navigation (namespace index)
+    adoc += '** ' + namespace.url + '\n';
+    namespace.pages.forEach(function (page) {
+      // namespace level pages
+      adoc += '*** xref:' + AntoraNavBaseDir + page.url + '.adoc' + '[' + page.url+ ']\n';
```

Comment:
I think separating the presentation name and the url into two fields would make it better since then you would retain the camelcase in the menus. So if page here has a page.url that is lowercase and a page.fullName that is the full namespace path like `tinymce.FakeClipboard` it would look a little better generated.

Reply:
ah yes i will implement this improvement, it should fix a bug where 
examplepage.html and ExamplePage.html are called

Reply:
Went with the second option, as there's the types as well that run into this and that's not defined by the template so I didn't want to touch that given it'd break other templates then. I also removed a fair bit of duplication with generating the type and definedBy links.